### ISSUE TITLE
fix ARD bug

### ISSUE THREAD

**Issue by Alwaysproblem** [Karl-Krauth/Sparse-GP#2]
the torch.mm is not the same as the Theano.dot. so I try to fix the problem.

the 216th in kernel.py should be substituted with:
tensordot_S_x_xl3 = torch.mm(S, x_xl3.contiguous().view(x_xl3.size()[0], -1))\
                                    .view(S.size()[0], x_xl3.size()[1], x_xl3.size()[-1])

lengthscale_gradient = -(torch.sum(D.t()[:, :, np.newaxis] * tensordot_S_x_xl3, dim=1) /
                                    length_scale ** 3)
